datafusion/functions-table/src/generate_series.rs

```diff
@@ -0,0 +1,180 @@
+            self._start, self.end, self.batch_size
+        )
+    }
+}
+
+impl LazyBatchGenerator for GenerateSeriesState {
+    fn generate_next_batch(&mut self) -> Result<Option<RecordBatch>> {
+        // Check if we've reached the end
+        if self.current > self.end {
+            return Ok(None);
+        }
+
+        // Construct batch
+        let batch_end = (self.current + self.batch_size as i64 - 1).min(self.end);
+        let array = Int64Array::from_iter_values(self.current..=batch_end);
+        let batch = RecordBatch::try_new(self.schema.clone(), vec![Arc::new(array)])?;
+
+        // Update current position for next batch
+        self.current = batch_end + 1;
+
+        Ok(Some(batch))
+    }
+}
+
+#[async_trait]
+impl TableProvider for GenerateSeriesTable {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.schema.clone()
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::Base
+    }
+
+    async fn scan(
+        &self,
+        state: &dyn Session,
+        _projection: Option<&Vec<usize>>,
+        _filters: &[Expr],
+        _limit: Option<usize>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        let batch_size = state.config_options().execution.batch_size;
+        match (self.start, self.end) {
+            (Some(start), Some(end)) => {
+                if start > end {
+                    return plan_err!(
+                        "End value must be greater than or equal to start value"
+                    );
+                }
+
+                Ok(Arc::new(LazyMemoryExec::try_new(
+                    self.schema.clone(),
+                    vec![Arc::new(RwLock::new(GenerateSeriesState {
+                        schema: self.schema.clone(),
+                        _start: start,
+                        end,
+                        current: start,
+                        batch_size,
+                    }))],
+                )?))
+            }
+            _ => {
+                // Either start or end is None, return a generator that outputs 0 rows
+                Ok(Arc::new(LazyMemoryExec::try_new(
+                    self.schema.clone(),
+                    vec![Arc::new(RwLock::new(GenerateSeriesState {
+                        schema: self.schema.clone(),
+                        _start: 0,
+                        end: 0,
+                        current: 1,
+                        batch_size,
+                    }))],
+                )?))
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct GenerateSeriesFunc {}
+
+impl TableFunctionImpl for GenerateSeriesFunc {
+    // Check input `exprs` type and number. Input validity check (e.g. start <= end)
+    // will be performed in `TableProvider::scan`
+    fn call(&self, exprs: &[Expr]) -> Result<Arc<dyn TableProvider>> {
+        // TODO: support 1 or 3 arguments following DuckDB:
```

Comment:
can we possibly file a ticket to track this?

Reply:
Tracked in https://github.com/apache/datafusion/issues/13615